src/main/java/io/cdap/plugin/gcp/gcs/source/GCSSource.java

```diff
@@ -248,6 +277,13 @@ public String getEncryptedMetadataSuffix() {
         DEFAULT_ENCRYPTED_METADATA_SUFFIX : encryptedMetadataSuffix;
     }
 
+    /**
+     * Extract bucket name out of path with the format: gs://<bucket>/path/to/output
+     */
+    public String getBucketName() {
+      return this.getPath().replace("gs://", "").split("/")[0];
```

Comment:
We have `GCSPath` class you can use to extract the bucket out

Reply:
+1, please use the GCSPath class for this.